components/invitation_modal/invitation_modal.tsx

```diff
@@ -0,0 +1,396 @@
+
+    channelsLoader = async (value: string) => {
+        if (!value) {
+            return this.props.invitableChannels;
+        }
+
+        this.debouncedSearchChannels(value);
+        return this.props.invitableChannels.filter((channel) => {
+            return channel.display_name.toLowerCase().startsWith(value.toLowerCase()) || channel.name.toLowerCase().startsWith(value.toLowerCase());
+        });
+    }
+
+    onChannelsChange = (channels: Channel[]) => {
+        this.setState((state) => ({
+            ...state,
+            invite: {
+                ...state.invite,
+                inviteChannels: {
+                    ...state.invite.inviteChannels,
+                    channels,
+                },
+            },
+        }));
+    }
+
+    onChannelsInputChange = (search: string) => {
+        this.setState((state) => ({
+            ...state,
+            invite: {
+                ...state.invite,
+                inviteChannels: {
+                    ...state.invite.inviteChannels,
+                    search,
+                },
+            },
+        }));
+    }
+
+    debouncedSearchProfiles = debounce((term: string, callback: (users: UserProfile[]) => void) => {
+        this.props.actions.searchProfiles(term).
+            then(({data}: {data: UserProfile[]}) => {
+                callback(data);
+                if (data.length === 0) {
+                    this.setState({termWithoutResults: term});
+                } else {
+                    this.setState({termWithoutResults: null});
+                }
+            }).
+            catch(() => {
+                callback([]);
+            });
+    }, 150);
+
+    usersLoader = (term: string, callback: (users: UserProfile[]) => void): Promise<UserProfile[]> | undefined => {
+        if (
+            this.state.termWithoutResults &&
+            term.startsWith(this.state.termWithoutResults)
+        ) {
+            callback([]);
+            return;
+        }
+        try {
+            this.debouncedSearchProfiles(term, callback);
+        } catch (error) {
+            callback([]);
+        }
+    };
+
+    onChangeUsersEmails = (usersEmails: Array<UserProfile | string>) => {
+        this.setState((state: State) => ({
+            ...state,
+            invite: {
+                ...state.invite,
+                usersEmails,
+            },
+        }));
+    }
+
+    onUsersInputChange = (usersEmailsSearch: string) => {
+        this.setState((state: State) => ({
+            ...state,
+            invite: {
+                ...state.invite,
+                usersEmailsSearch,
+            },
+        }));
+    }
+
+    getBackdrop = () => {
+        // 'static' means backdrop clicks do not close
+        // true means backdrop clicks do close
+        // false means no backdrop
+        if (this.state.view === View.RESULT || (!this.props.canAddUsers && !this.props.canInviteGuests) || this.state.invite.usersEmailsSearch === 'nopermissions') {
```

Comment:
This and other `'nopermissions'` related code is a hack to make it easy to see this view for testing purposes. It should be removed prior to merging.

Reply:
Not sure how it would be in that state, either, but wanted these protections in case something goes wrong with the guards.